guix-science-nonfree/packages/gurobi.scm

```diff
@@ -0,0 +1,73 @@
+;; -*- mode: scheme; eval: (guix-devel-mode 1); geiser-scheme-implementation: guile -*-
+
+;;;
+;;; Copyright © 2021 Quantile Technologies <phil.beadling@quantile.com>
+;;;
+;;; This program is free software; you can redistribute it and/or modify it
+;;; under the terms of the GNU General Public License as published by
+;;; the Free Software Foundation; either version 3 of the License, or (at
+;;; your option) any later version.
+;;;
+;;; This program is distributed in the hope that it will be useful, but
+;;; WITHOUT ANY WARRANTY; without even the implied warranty of
+;;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;;; GNU General Public License for more details.
+;;;
+;;; You should have received a copy of the GNU General Public License
+;;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+(define-module (guix-science-nonfree packages gurobi)
+  #:use-module (guix build-system python)
+  #:use-module (guix packages)
+  #:use-module (guix download) ;; url-fetch
+  #:use-module (guix git) ;; git-checkout
+  #:use-module (gnu packages elf)) ;; patchelf
+
+(define-public python-gurobipy
+  (package
+    (name "python-gurobipy")
+    (version "9.0.1")
+    (source (origin
+              (method url-fetch)
+              (uri (string-append "https://packages.gurobi.com/"
+				  (substring version 0 3) "/gurobi"
+				  version "_linux64.tar.gz"))
+              (sha256
+               (base32
+                "1dvcz0ph17y7dlsn2z5akzhbnc32f6wd1ppfc7p0w60il76zmqhp"))))
+    (build-system python-build-system)
+    (native-inputs
+     `(("patchelf" ,patchelf)))
+    (arguments ;;(gurobi-package-arguments version))
+     `(#:use-setuptools? #f ;; distuils package
+			 #:tests? #f ;; no tests in package
+			 #:phases (modify-phases %standard-phases
+				    (add-after 'unpack 'cd-to-source-dir
+				      (lambda _ (chdir "linux64") #t))
+				    ;; copy and symlink gurobi binary and adjust python lib rpath
+				    ;; to look for it in it's own directory.
+				    (add-after 'install 'install-gurobi-library
+				      (lambda* (#:key outputs #:allow-other-keys)
+					(let* ((dir (string-append (assoc-ref outputs "out")
+								   "/lib/python3.8/site-packages/gurobipy/"))
```

Comment:
Can we use `site-packages` from `(guix build python-build-system)` here to avoid breakage as soon as Python is updated?